gn/build_overrides/chip.gni

```diff
@@ -15,4 +15,5 @@
 declare_args() {
   # Root directory for CHIP.
   chip_root = "//"
+  chip_platform_root = "${chip_root}/src/platform/"
```

Comment:
Should just use an argument for this

```
declare_args() {
  # Use an external target for platform.
  chip_platform_target = ""
}
```

The build_overrides is mostly for finding repositories because it needs to be imported without knowing where the .gni file is (it's in the repository..). If we know where the chip repository is, we can find src/platform/device.gni, so we can just import that and use declare_args().